WooCommerce/Classes/ViewRelated/ReusableViews/TwoColumnHeadlineAttributedFootnoteTableViewCell.swift

```diff
@@ -0,0 +1,77 @@
+import UIKit
+
+
+/// Represents a cell with a two-column title "row"
+/// and an attributed string footnote "row" below the titles
+///
+final class TwoColumnHeadlineAttributedFootnoteTableViewCell: UITableViewCell {
+
+    /// We want this reusable cell to have a consistent appearance
+    /// everywhere it's used, so we closed access to the labels.
+    ///
+    @IBOutlet private weak var leftTitleLabel: UILabel!
+    @IBOutlet private weak var rightTitleLabel: UILabel!
+    @IBOutlet private weak var footnoteLabel: UILabel!
+
+    // Expose the properties of a label that should be available for configuration.
+
+    /// Left title label text
+    ///
+    var leftText: String? {
+        get {
+            return leftTitleLabel?.text
+        }
+        set {
+            leftTitleLabel?.text = newValue
+        }
+    }
+
+    /// Right title label text
+    ///
+    var rightText: String? {
+        get {
+            return rightTitleLabel?.text
+        }
+        set {
+            rightTitleLabel?.text = newValue
+        }
+    }
+
+    /// Footnote label text
+    ///
+    var footnoteAttributedText: NSAttributedString? {
+        get {
+            return footnoteLabel?.attributedText
+        }
+        set {
+            footnoteLabel?.attributedText = newValue
+        }
+    }
```

Comment:
Originally, I publicly exposed the footerLabel in `TwoColumnHeadlineFooterTableViewCell`. But then I realized it violates the single responsibility principle and the open / closed principle, because I would have to open the `footnoteLabel` for public access and create the new `footnoteAttributedText` variable. Now footnoteLabel has 2 responsibilities. Either display text or display attributed text.

Adding a new var `footnoteAttributedText` doesn't work by itself, because I have to check if the `footnoteLabel.text` property is nil, remove the original label styling, and apply a different label style. All of those actions should be done in footnoteAttributedText's `didSet` method, but `didSet` can't be used since there is already a getter.

I created a new tableviewcell, `TwoColumnHeadlineAttributedFooterTableViewCell` as a solution to all of these problems.

Reply:
I think it's pretty common to have multiple ways of styling a view for similar use cases, the name "TwoColumnHeadlineFooterTableViewCell" feels pretty generic and I wouldn't be surprised if there is more than one way to set the headline/footer.

if just adding the same code
```
var footnoteAttributedText: NSAttributedString? {
        get {
            return footnoteLabel?.attributedText
        }
        set {
            footnoteLabel?.attributedText = newValue
        }
}
```

to `TwoColumnHeadlineFootnoteTableViewCell` and using it for the Refund cells, I feel that might work?

Regarding the getters, I checked the Apple doc:

About [`UILabel`'s `text` property](https://developer.apple.com/documentation/uikit/uilabel/1620538-text):
> Assigning a new value to this property also replaces the value of the attributedText property with the same text, although without any inherent style attributes.

About [`UILabel`'s `attributedText ` property](https://developer.apple.com/documentation/uikit/uilabel/1620542-attributedtext):
> Assigning a new value to this property also replaces the value of the text property with the same string data, although without any formatting information. In addition, assigning a new a value updates the values in the font, textColor, and other style-related properties so that they reflect the style information starting at location 0 in the attributed string.

From the docs above, it seems like setting either `text` or `attributedText` will update the label's text with or without styles. So I don't think it'd require the knowledge of the other property. Also, the getters aren't really used and the setters could also be in a function form like `func updateFootnoteText(text: String?)` / `func updateFootnoteAttributedText(footnoteAttributedText: NSAttributedString?)`

My concern for creating another cell with the same layout/style with the exception of a setter is that whenever we update the layout or style in one cell, we'd have to remember to update the other cell and not every dev might be aware of this.

Reply:
I disagree. There is less cost to updating the layout and style in a separate cell than there is with maintaining a single cell that does too many things and will need refactored in the future. @pmusolino would you be a tie-breaker? I believe both @jaclync and I would agree that we will accept whatever decision you make.

Reply:
Edit: I apologize for the misunderstanding. The whole point of the new cell was to **remove** the `.footerText` variable so the cell was not doing two things. I have edited this new cell to demonstrate what I was originally thinking.

Reply:
To be honest, since the meaning of these two cells is the same, I would be more inclined to use a single cell, maybe exposing two configure methods that allow passing a text or an attributed text.

The SOLID principles are closer to guidelines than rules; they’re impossible to follow 100% of the time, and yet, adherence to 90% of the principles will ultimately serve to create a far superior application. In that way, it’s okay to break the Single Responsibility Principle in this way, and I do without feeling guilty. This is one of these cases, in my opinion, where readability and less code duplication has a better implication than SRP.

Reply:
>In that way, it’s okay to break the Single Responsibility Principle in this way, and I do without feeling guilty

🙂 Thanks Paolo. I'll change it back to using the original cell and delete the code for this one.